### PR TITLE
Fix testMapperDynamicIndexSetting() failures

### PR DESCRIPTION
**Pull request by martijnvg**
Currently these tests run against any old cluster older than 8.0.0, but the fix that allowed `index.mapper.dynamic` to exist is only available in 7.17.22.

Adjust these tests to only run if old cluster is after version 7.17.21 and before 8.0.0

### CODE REVIEW

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/UpgradeWithOldIndexSettingsIT.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public UpgradeWithOldIndexSettingsIT(@Name("cluster") FullClusterRestartUpgradeS
     public void testMapperDynamicIndexSetting() throws IOException {
         assumeTrue(
             "Setting deprecated in 6.x, but remained in 7.x and is no longer defined in 8.x",
```
Comment (kkrik-es) [inline qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/UpgradeWithOldIndexSettingsIT.java:53]: Update the comment about the 7.17.21 part.

**Review by kkrik-es — APPROVED** [review]

**Review by benwtrent — APPROVED** [review]
I say :shipit:

**Comment by benwtrent** [comment]
run elasticsearch-ci/8.13.4 / bwc

**Comment by benwtrent** [comment]
run elasticsearch-ci/8.13.4 / bwc

**Pull request merged**